File: application/server/src/test/kotlin/io/github/siyual_park/application/server/dummy/DummyCreateUserRequest.kt
package io.github.siyual_park.application.server.dummy

import io.github.siyual_park.application.server.dto.request.CreateUserRequest
import io.github.siyual_park.test.DummyEmailFactory
import io.github.siyual_park.test.DummyNameFactory
import io.github.siyual_park.test.DummyStringFactory
import io.github.siyual_park.test.resolveNotNull
import java.util.Optional

object DummyCreateUserRequest {
    data class Template(
        val name: Optional<String>? = null,
        val email: Optional<String>? = null,
        val password: Optional<String>? = null,
    )

    fun create(template: Template? = null): CreateUserRequest {
        return CreateUserRequest(
            name = resolveNotNull(template?.name) { DummyNameFactory.create(10) },
            email = resolveNotNull(template?.email) { DummyEmailFactory.create(15) },
            password = resolveNotNull(template?.password) { DummyStringFactory.create(10) },
        )
    }
}

File: k6-tests/src/dummy/dummy-create-user-request.ts
// @ts-ignore
import { randomString } from 'https://jslib.k6.io/k6-utils/1.2.0/index.js';

import { CreateUserRequest } from '../request';

function dummyCreateUserRequest(): CreateUserRequest {
  return {
    name: randomString(10),
    email: `${randomString(10)}@test.com`,
    password: randomString(10),
  };
}

export default dummyCreateUserRequest;
